components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -254,13 +254,68 @@ public Response getRegistryEntryByUuid(String registryId, String entryId, Messag
     @Override
     public Response updateRegistryEntry(String registryId, String entryId, RegistryEntryDTO registryEntry, InputStream
             definitionFileInputStream, Attachment definitionFileDetail, MessageContext messageContext) {
+        String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+        String user = RestApiUtil.getLoggedInUsername();
+        EndpointRegistry registryProvider = new EndpointRegistryImpl();
+        EndpointRegistryEntry updatedEntry = null;
+
+        try {
+            EndpointRegistryInfo endpointRegistry =
+                    registryProvider.getEndpointRegistryByUUID(registryId, tenantDomain);
+            if (endpointRegistry == null) {
+                RestApiUtil.handleResourceNotFoundError("Endpoint registry with the id: " + registryId +
+                        " is not found", log);
+            }
+            EndpointRegistryEntry endpointRegistryEntry =
+                    registryProvider.getEndpointRegistryEntryByUUID(registryId, entryId);
+            if (endpointRegistryEntry == null) {
+                RestApiUtil.handleResourceNotFoundError("Endpoint registry entry with the id: " + entryId +
+                        " is not found", log);
+            }
 
-        return Response.ok().entity(registryEntry).build();
+            ResourceFile definitionFile = new ResourceFile(definitionFileInputStream, definitionFileDetail
```

Comment:
can any null pointers occur here when we don't upload a file?

Reply:
yeah it occurs. I will fix it. If there is no file attached, still we should update the other details right?

Reply:
yeah, let's keep the existing blob in the DB as it is and update the rest in that case.

Reply:
we should also think of the use case where the user wants to remove the blob from the DB and update. in that sense, we should set an empty blob if no file is attached.

Reply:
since there would be no way to achieve the above if we don't update the blob, let's go ahead and set an empty blob if no file is attached. wdyt

Reply:
Fixed